### PR TITLE
Add rector

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Install [rector](https://getrector.com/)

> Kudos for the amazing project, @TomasVotruba! 🥳 

## 🔖 Changes

- Apply multiple rector set rules to modernise the overall code agreements and style

### CODE REVIEW

File: src/Console/Domain/AllAppModules/AllAppModulesFinder.php [hunk 1]
```diff
@@ -27,11 +27,12 @@ public function findAllAppModules(string $filter): array
         /** @var SplFileInfo $fileInfo */
         foreach ($this->fileIterator as $fileInfo) {
             $appModule = $this->createAppModule($fileInfo, $filter);
-            if ($appModule !== null && $this->isFacade($appModule)) {
+            if ($appModule instanceof \Gacela\Console\Domain\AllAppModules\AppModule && $this->isFacade($appModule)) {
```
Comment (JesusValera) [inline src/Console/Domain/AllAppModules/AllAppModulesFinder.php:30]: FQN here

File: src/Framework/Bootstrap/SetupGacela.php [hunk 2]
```diff
@@ -181,7 +181,7 @@ public function buildAppConfig(AppConfigBuilder $builder): AppConfigBuilder
     {
         $builder = parent::buildAppConfig($builder);
 
-        if ($this->appConfigBuilder) {
+        if ($this->appConfigBuilder instanceof \Gacela\Framework\Config\GacelaConfigBuilder\AppConfigBuilder) {
```
Comment (JesusValera) [inline src/Framework/Bootstrap/SetupGacela.php:184]: FQN here too :D

File: src/Framework/Bootstrap/SetupGacela.php [hunk 3]
```diff
@@ -211,7 +211,7 @@ public function buildBindings(
     ): BindingsBuilder {
         $builder = parent::buildBindings($builder, $externalServices);
 
-        if ($this->bindingsBuilder) {
+        if ($this->bindingsBuilder instanceof \Gacela\Framework\Config\GacelaConfigBuilder\BindingsBuilder) {
```
Comment (JesusValera) [inline src/Framework/Bootstrap/SetupGacela.php:214]: Here

File: src/Framework/Bootstrap/SetupGacela.php [hunk 4]
```diff
@@ -240,7 +240,7 @@ public function buildSuffixTypes(SuffixTypesBuilder $builder): SuffixTypesBuilde
     {
         $builder = parent::buildSuffixTypes($builder);
 
-        if ($this->suffixTypesBuilder) {
+        if ($this->suffixTypesBuilder instanceof \Gacela\Framework\Config\GacelaConfigBuilder\SuffixTypesBuilder) {
```
Comment (JesusValera) [inline src/Framework/Bootstrap/SetupGacela.php:243]: And here

File: src/Framework/ClassResolver/AbstractClassResolver.php [hunk 5]
```diff
@@ -154,7 +154,7 @@ private function createInstance(string $resolvedClassName): object
 
     private function getGacelaConfigFile(): GacelaConfigFileInterface
     {
-        if ($this->gacelaFileConfig === null) {
+        if (!$this->gacelaFileConfig instanceof \Gacela\Framework\Config\GacelaFileConfig\GacelaConfigFileInterface) {
```
Comment (JesusValera) [inline src/Framework/ClassResolver/AbstractClassResolver.php:157]: And here

**Review by JesusValera — APPROVED** [review]
<3

File: src/Console/Domain/AllAppModules/AllAppModulesFinder.php (hunk 1 above)
Comment (Chemaclass) [reply src/Console/Domain/AllAppModules/AllAppModulesFinder.php:30]: Thanks! I will remove them

**Pull request merged**